### PR TITLE
Design doc for bring-your-own-host

### PR DESCRIPTION
**Pull request by 4everming**
### What this PR does / why we need it
Introduce document to describe the plan to support bring-your-own-host provider in tanzu-framework.

### Which issue(s) this PR fixes

Related to: #823

### PR Checklist
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: docs/design/bringyourownhost-design.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Bring your own host enhancement design
```
Comment (vuil) [inline docs/design/bringyourownhost-design.md:1]:
```suggestion
# Bring-Your-Own-Host Design
```

File: docs/design/bringyourownhost-design.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+
+## Abstract
+
+The Bring Your Own Host aka BYOH aims at enabling Kubernetes cluster life cycle management on user provided hosts (most of the case are bare metal hosts).
```
Comment (vuil) [inline docs/design/bringyourownhost-design.md:7]:
```suggestion
The Bring Your Own Host aka BYOH aims at enabling Kubernetes cluster life cycle management on user-provided hosts (bare metal hosts in most cases).
```

File: docs/design/bringyourownhost-design.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+But currently only 1 infrastructure provider is supported in tanzu-framework,  this is to enhance current tanzu-framework so that users can use tanzu CLI to create BYOH workload cluster.
```
Comment (vuil) [inline docs/design/bringyourownhost-design.md:9]: I am not sure I understand this. Can you clarify?

**Review by vuil — COMMENTED** [review]
Thanks for the addition. This has the potential of opening up the CLI to logs of other deployment scenarios!

Will take another pass at this later, but here are some inital comments and minor language suggestions.

**Review by vuil — APPROVED** [review]

**Review by vuil — APPROVED** [review]
Thanks for updating the design doc. It is a lot clearer now what the current and upcoming phases will set out to accomplish. I took the liberty of clarifying the title and linking back to the issue filed.

The doc lgtm, but it needs some tweaking to conform to the markdown linting we are doing on all markdown files. Please take a look at the issues raised by the lint checks and address them.

Details on what is expected is also mentioned in the output of the Lint checker logs, but repeated here:
https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md

**Comment by 4everming** [comment]
> Thanks for updating the design doc. It is a lot clearer now what the current and upcoming phases will set out to accomplish. I took the liberty of clarifying the title and linking back to the issue filed.
> 
> The doc lgtm, but it needs some tweaking to conform to the markdown linting we are doing on all markdown files. Please take a look at the issues raised by the lint checks and address them.
> 
> Details on what is expected is also mentioned in the output of the Lint checker logs, but repeated here: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md

Thanks Vui for let me know that

**Pull request merged**